python/cuml/common/base.pyx

```diff
@@ -282,6 +282,24 @@ class Base:
             else:
                 raise AttributeError
 
+    def _set_base_attributes(self, X, output_type=False, y=None,
+                             n_features=None):
```

Comment:
This method is likely to be extended in the future, I think doing one line per parameter would be a good idea